libsolidity/codegen/CompilerContext.cpp

```diff
@@ -227,6 +227,13 @@ void CompilerContext::injectVersionStampIntoSub(size_t _subIndex)
 	sub.injectStart(fromBigEndian<u256>(binaryVersion()));
 }
 
+void CompilerContext::injectMetadataHashIntoSub(size_t _subIndex, h256 const& _metadataHash)
+{
+	eth::Assembly& sub = m_asm->sub(_subIndex);
+	sub.injectStart(Instruction::POP);
+	sub.injectStart(u256(_metadataHash));
```

Comment:
Good find! It might be quite hard to force that to `push32`. Would it be better to change the documentation/specification?
On the other hand, if we add that to the end of the bytecode, then there might be another way to do it.